Formula/clickhouse.rb

```diff
@@ -0,0 +1,74 @@
+class Clickhouse < Formula
+  desc "Free analytics DBMS for big data with SQL interface"
+  homepage "https://clickhouse.tech"
+  url "https://github.com/ClickHouse/ClickHouse/releases/download/v21.6.3.14-stable/ClickHouse_sources_with_submodules.tar.gz"
+  version "21.6"
+  sha256 "696bbcdef4796bfdf1becdb78abb5e97c0220acc9863f5a821193dd297ce428b"
+  license "Apache-2.0"
+  head "https://github.com/ClickHouse/ClickHouse.git"
+  depends_on "cmake" => :build
+  depends_on "ninja" => :build
+  depends_on xcode: :build
+
+  def install
+    mkdir_p bin
+    mkdir_p "#{HOMEBREW_PREFIX}/etc/clickhouse-server"
+    mkdir_p "#{var}/log/clickhouse-server"
+    mkdir_p "#{var}/lib/clickhouse-server"
+    mkdir_p "#{var}/run/clickhouse-server"
+
+    inreplace "programs/server/config.xml" do |s|
+      s.gsub! "<!-- <max_open_files>262144</max_open_files> -->", "<max_open_files>262144</max_open_files>"
+    end
+    inreplace "cmake/warnings.cmake" do |s|
+      s.gsub!(/add_warning\(frame-larger-than=(\d*)\)/, "add_warning(frame-larger-than=131072)")
+    end
+
+    args = std_cmake_args
+    args.delete("-DCMAKE_BUILD_TYPE=Release")
+    args << "-DCMAKE_BUILD_TYPE=RelWithDebInfo"
+
+    system "cmake", ".", *args
+    system "ninja", "clickhouse-bundle"
+
+    system "#{buildpath}/programs/clickhouse", "install",
+                                               "--binary-path=#{bin}",
+                                               "--config-path=#{HOMEBREW_PREFIX}/etc/clickhouse-server",
+                                               "--log-path=#{var}/log/clickhouse-server",
+                                               "--data-path=#{var}/lib/clickhouse-server",
+                                               "--pid-path=#{var}/run/clickhouse-server"
+  end
+
+  plist_options manual: "clickhouse-server --config-file #{HOMEBREW_PREFIX}/etc/clickhouse-server/config.xml"
+
+  def plist
```

Comment:
Can we use a `service` block instead of manually creating a `plist`?

Reply:
I guess sure we can. But I can't find any documentation about this and probably try to catch the idea from  https://github.com/Homebrew/homebrew-core/pull/79366